### ISSUE TITLE
the meaning of these opration of std and mean?

### ISSUE THREAD

**Issue by Litt1eMarsun** [neu-vi/OmniControl#28]
I noticed that the author repeatedly uses normalization and inverse normalization in the paper's code. During the sampling process, this is specifically reflected as follows:

1. In the process of handling hints in `OmniControl/utils/text_control_example.py`:
```python
joints = (joints - raw_mean.reshape(1, 22, 3)) / raw_std.reshape(1, 22, 3)
```

2. In the gradient calculation function of spatial guidance in `OmniControl/diffusion/gaussian_diffusion.py`:
```python
def gradients(self, x, hint, mask_hint, joint_ids=None):
    with torch.enable_grad():
        x.requires_grad_(True)
        x_ = x.permute(0, 3, 2, 1).contiguous()
        x_ = x_.squeeze(2)
        x_ = x_ * self.std + self.mean 
```

3. In the guidance function of spatial guidance in `OmniControl/diffusion/gaussian_diffusion.py`:
```python
if self.raw_std.device != hint.device: 
    self.raw_mean = self.raw_mean.to(hint.device)
    self.raw_std = self.raw_std.to(hint.device)
    self.mean = self.mean.to(hint.device)
    self.std = self.std.to(hint.device)
hint = hint * self.raw_std + self.raw_mean  #
```

4. In the `plot_3d_motion` function for visualization:
...

What is the reason for these operations?

**Comment by ymingxie** [comment]
We follow MDM’s implementation (see [code link](https://github.com/GuyTevet/motion-diffusion-model/blob/main/data_loaders/humanml/data/dataset.py#L366)), which normalizes joint representations before feeding them into the model. Accordingly, during control, we also normalize the control signals using the pre-computed mean and standard deviation to ensure consistency with the normalized human joint data.

The normalized data is easier for the model to learn during training.
